### PR TITLE
Fomalhaut MRL

### PR DESCRIPTION
**Pull request by Gannets**
## About the PR 
Adds a currently unatainable 6 shot homing missle launcher.

https://user-images.githubusercontent.com/8319565/190931531-0ad9f3e3-6c4e-4cef-8c93-36446feb8471.mp4

## Why's this needed? 
Crimes.

## Changelog

```changelog
(u)Gannets
(*)Added the currently unattainable Fomalhaut MRL, 6-shot homing missile launcher. Homing projectile code by Zjdtmkhzt!
```

### CODE REVIEW

File: code/obj/item/gun/ammo.dm [hunk 1]
```diff
@@ -1026,6 +1026,18 @@ ABSTRACT_TYPE(/obj/item/ammo/bullets/pipeshot)
...
 	sound_load = 'sound/weapons/gunload_mprt.ogg'
 
+/obj/item/ammo/bullets/mrls
+	sname = "MRLS rocket pack"
+	name = "MRLS rocket pack"
+	amount_left = 6
+	max_amount = 6
+	icon_state = "rpg_rocket"
+	ammo_type = new /datum/projectile/bullet/rpg/homing/mrls
+	ammo_cat = AMMO_ROCKET_MRLS
```
Comment (Tarmunora) [inline code/obj/item/gun/ammo.dm:1036]: need to add the ammo cat define to `_std/defines/projectiles.dm`

File: code/obj/item/gun/kinetic.dm [hunk 2]
```diff
@@ -1392,6 +1392,47 @@ ABSTRACT_TYPE(/obj/item/gun/kinetic)
 			src.UpdateIcon()
 			return
 
+/obj/item/gun/kinetic/mrls
+	desc = "A multiple rocket launch system."
+	name = "\improper MRLS-6"
+	icon = 'icons/obj/large/64x32.dmi'
+	inhand_image_icon = 'icons/mob/inhand/hand_weapons.dmi'
+	icon_state = "rpg7"
+	item_state = "rpg7"
```
Comment (Tarmunora) [inline code/obj/item/gun/kinetic.dm:1401]:
```suggestion
	icon_state = "mrls"
	item_state = "mrls"
```

File: code/obj/item/gun/kinetic.dm [hunk 3]
```diff
@@ -1392,6 +1392,47 @@ ABSTRACT_TYPE(/obj/item/gun/kinetic)
...
+	wear_image_icon = 'icons/mob/clothing/back.dmi'
+	flags = ONBACK
+	w_class = W_CLASS_BULKY
+	throw_speed = 2
+	throw_range = 4
+	force = MELEE_DMG_LARGE
+	contraband = 8
+	ammo_cats = list(AMMO_ROCKET_ALL)
```
Comment (Tarmunora) [inline code/obj/item/gun/kinetic.dm:1409]: we probably *don't* want people to be able to load this with 6 full-strength RPG rockets (and then rapid-fire them etc)

File: code/obj/item/gun/kinetic.dm [hunk 4]
```diff
@@ -1392,6 +1392,47 @@ ABSTRACT_TYPE(/obj/item/gun/kinetic)
...
+		..()
+
+	disposing()
+		STOP_TRACKING_CAT(TR_CAT_NUKE_OP_STYLE)
+		..()
+
+	loaded
+		New()
+			..()
+			ammo.amount_left = 1
```
Comment (Tarmunora) [inline code/obj/item/gun/kinetic.dm:1433]:
```suggestion
			ammo.amount_left = 6
```

File: code/modules/projectiles/bullet.dm [hunk 5]
```diff
@@ -1313,6 +1313,98 @@ datum/projectile/bullet/autocannon
 			explosion_new(null, T, 36, 0.45)
 		return
 
+/datum/projectile/bullet/rpg/homing
```
Comment (Tarmunora) [inline code/modules/projectiles/bullet.dm:1316]: this probably should not be a child of `/datum/projectile/bullet/rpg`

**Review by Tarmunora — CHANGES_REQUESTED** [review]
also I'm not entirely sure, but I think it's MLRS, not MRLS 🤔

**Comment by NightmareChamillian** [comment]
Honestly, this looks incredibly fun- with how slow the rockets move, I can imagine it supporting some frankly comedic chase sequences. Shame it'll never spawn during normal gamemode conditions, though.

Also, I checked, and it is MLRS.

**Pull request merged**